app/src/main/java/app/revanced/integrations/youtube/patches/components/LayoutComponentsFilter.java

```diff
@@ -298,6 +307,10 @@ boolean isFiltered(@Nullable String identifier, String path, byte[] protobufBuff
         // TODO: This also hides the feed Shorts shelf header
         if (matchedGroup == searchResultShelfHeader && contentIndex != 0) return false;
 
+        if (contentIndex == 0 && matchedGroup == horizontalShelves && hideShelves()) {
```

Comment:
I am not entirely sure if this works this way and those shelves don't show up for 
 me @LisoUseInAIKyrios perhaps you know?